LiteCore/tests/N1QLParserTest.cc

```diff
@@ -103,13 +103,15 @@ TEST_CASE_METHOD(N1QLParserTest, "N1QL properties", "[Query][N1QL][C]") {
     CHECK(translate("select `mr.grieves`.`hey`") == "{'WHAT':[['.mr\\\\.grieves.hey']]}");
     CHECK(translate("select `$type`") == "{'WHAT':[['.\\\\$type']]}");
 
-    CHECK(translate("select meta.id") == "{'WHAT':[['._id']]}");
-    CHECK(translate("select meta.sequence") == "{'WHAT':[['._sequence']]}");
-    CHECK(translate("select meta.deleted") == "{'WHAT':[['._deleted']]}");
-    CHECK(translate("select db.meta.id") == "{'WHAT':[['.db._id']]}");
-    CHECK(translate("select meta.bogus") == "");    // only specific meta properties allowed
-    CHECK(translate("select db.meta.bogus") == "");
-
+    CHECK(translate("select meta().id") == "{'WHAT':[['_.',['meta'],'.id']]}");
+    CHECK(translate("select meta().sequence") == "{'WHAT':[['_.',['meta'],'.sequence']]}");
+    CHECK(translate("select meta().deleted") == "{'WHAT':[['_.',['meta'],'.deleted']]}");
+    CHECK(translate("select meta(db).id from db") == "{'FROM':[{'AS':'db'}],'WHAT':[['_.',['meta','db'],'.id']]}");
+    {
+        ExpectingExceptions x;
+        CHECK_THROWS_WITH(translate("select meta().bogus"), "'bogus' is not a valid Meta key");
+        CHECK_THROWS_WITH(translate("select meta(db).bogus from db"), "'bogus' is not a valid Meta key");
```

Comment:
Is this a change in behavior with exceptions?

Reply:
Yes. In the deprecated method, we check the meta key in LEG parser. The error shows as empty string. In the new method, we check it in QueryParser. "meta().bogus" is valid w.r.t. LEG.

Reply:
I think it should be ok as long as we throw exceptions elsewhere.  I'm concerned about how they will present themselves.  They would need to make it up to the C API boundary and get trapped into a `C4Error`

Reply:
It behaves same as this one, 
    // "custId" is implicitly scoped by the unique alias, "orders".
    CHECK(translate("SELECT DISTINCT custId FROM orders where test_id = 'agg_func' ORDER BY custId") ==
          "{'DISTINCT':true,'FROM':[{'AS':'orders'}],'ORDER_BY':[['.custId']],"
          "'WHAT':[['.custId']],'WHERE':['=',['.test_id'],'agg_func']}");
    {
        ExpectingExceptions x;
        CHECK_THROWS_WITH(translate("SELECT custId, other.custId FROM orders JOIN orders other "
                                    "ON orders.test_id = other.test_id ORDER BY custId"),
                          "property 'custId.' does not begin with a declared 'AS' alias");
    }

Reply:
There are numerous occasions that QueryParser will throw. It's a question we would have to address.

Reply:
If the query parser already throws then it is likely that it does the right thing already.  I imagine it would bubble into a call to either `c4query_run` or `c4query_new` and be captured there.